### ISSUE TITLE
Github Actions example

### ISSUE THREAD

**Issue by xHyroM** [bikecoders/ngx-deploy-npm#540]
Hey, i'm trying to setup this with github actions but im stucked on npm credentials.
https://github.com/xHyroM/things/blob/main/.github/workflows/release.yml

I know, its recommended to use https://github.com/jscutlery/semver and i have installed it, but want to figure this out and then somehow merge it. I'm really new using nx and don't know with it so much.

Some example of using github actions would be great and also with jscutlery's semver

**Comment by xHyroM** [comment]
> Currently, there is an [Open Issue](https://github.com/bikecoders/ngx-deploy-npm/issues/245) to include an example with Github Actions on the readme.
> 
> We use semver and ngx-deploy-npm (yes, we are using ourselves to publish our package to NPM). You can always go and [see a production like examples](https://github.com/bikecoders/ngx-deploy-npm/blob/main/.github/workflows/publishment.yml) of both technologies in action

I saw this production example, but confused what `nx version` exactly does. Does it increase semver version by jscutlery's semver and publish to npm?

**Comment by dianjuar** [comment]
> Does it increase semver version by jscutlery's semver and publish to npm

Yes. It creates a version and after that, tries to publish that version to NPM.

**Comment by dianjuar** [comment]
This might help you, a generic `project.json` with semver and ngx-deploy-npm

```json
{
  "name": "YOUR_LIB_NAME",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/YOUR_LIB_NAME/src",
  "projectType": "library",
  "targets": {
    "deploy": {
      "executor": "ngx-deploy-npm:deploy",
      "options": {
        "access": "public"
      }
    },
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "postTargets": ["YOUR_LIB_NAME:deploy"],
        "versionTagPrefix": "v"
      }
    }
  },
  "tags": []
}
```

To publish you must run `nx version YOUR_LIB_NAME`.

# Explanation

To publish a new version, you run the `version` target. The version target has an option called `postTargets`, which is what the target will run after the version is created. You might wonder, why do it in that way? the reason is that you might not have a new version to publish every time you run `version`. When that happens, the `version` target will not activate the deployment. That is really convenient because if you run the target `deploy` when there is no version to publish, the target `deploy` will fail because you are trying to publish an already published version.

For instance, since your last deployment, there are only `ci` and `test` commits there is no new version to publish.

**Comment by dianjuar** [comment]
> What about monorepo with more packages?

What I'm seeing other devs are using is creating the same `version` target in all the projects, and they run the version target on all the affected projects to verify if the changes make a new version. 

The command would be `nx affected --target=version`
